agenda/presentation/src/main/java/com/gkp/agenda/presentation/home/AgendaUiState.kt

```diff
@@ -16,4 +20,26 @@ data class AgendaUiState(
 
     val uiDates: List<DateWithSelected>
         get() = getCurrentAndNextSixWeekDayDates(selectedDate)
+
+    val uiFormattedSelectedDate: String
+        @Composable
+        get() = when (selectedDate) {
```

Comment:
Composable functions are tied to the Compose runtime and its lifecycle and I am not sure how this would behave when used inside a data class like this 🤔 This annotation also ties this data class directly to Compose, which makes it very coupled and makes testing harder.

See my comment on the `UIText` interface and see how you can use it to modify this class not to be tied to Compose directly.

Reply:
I'm not 100% sure, but coupling this to Compose might prevent you from writing JVM unit tests for your ViewModel (that uses this state class). JVM unit tests may have no references to the Android SDK - now I'm not sure if Compose falls under that, but either way it does couple the state to Compose, that's true. I'd also prefer the UiText approach from below